### PR TITLE
Add Neural Search section

### LINKED ISSUE #1563: [DOC] Documentation for new Neural Search OpenSearch plugin
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [X] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Add documentation for new neural search plugin. More details on plugin can be found in [RFC](https://github.com/opensearch-project/neural-search/issues/11). Plugin uses ml-commons deep learning model inference support to produce dense vectors from text and then uses the k-NN plugin to make the vectors searchable and/or produce k-NN queries. 

In the initial release, we are introducing a new ingestion processor (text_embedding) and query type (neural)

Release targeted as experimental for 2.4.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
1. Neural Search RFC(APIs/Usage/more details): https://github.com/opensearch-project/neural-search/issues/11
2. ml-commons related issue: https://github.com/opensearch-project/documentation-website/issues/1486

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Signed-off-by: Naarcha-AWS <naarcha@amazon.com>

Closes #1563 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _neural-search-plugin/index.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+---
+layout: default
+title: Neural Search plugin
+nav_order: 1
+has_children: false
+has_toc: false
+---
+
+The OpenSearch Neural Search plugin enables the integration of Machine Learning (ML) language models into their search workloads. Because language models can sometime require a over 100mb of memory, the Neural Search plugin automatically turns your index documents into vectors and ingests your vectorized documents into an OpenSearch index.
```
Comment (jmazanec15) [inline _neural-search-plugin/index.md:9]: Suggestion: The OpenSearch Neural Search plugin enables the integration of Machine Learning (ML) language models into user's search workloads. During ingestion and search, users can provide text that the neural search plugin will automatically transform into vectors to be used in vector search.

File: _neural-search-plugin/index.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
+---
+layout: default
+title: Neural Search plugin
+nav_order: 1
+has_children: false
+has_toc: false
+---
+
```
Comment (jmazanec15) [inline _neural-search-plugin/index.md:8]: Also, we need some kind of label that says it is experimental.

File: _neural-search-plugin/index.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+The OpenSearch Neural Search plugin enables the integration of Machine Learning (ML) language models into their search workloads. Because language models can sometime require a over 100mb of memory, the Neural Search plugin automatically turns your index documents into vectors and ingests your vectorized documents into an OpenSearch index.
+
+## Install the Neural Search plugin
```
[the remark this answers is not in the document]
Comment (Naarcha-AWS) [reply _neural-search-plugin/index.md:11]: So all user's need to do is install the tar package?

[the remark this answers is not in the document]
Comment (jmazanec15) [reply _neural-search-plugin/index.md:11]: Correct

File: _opensearch/install/plugins.md [hunk 4]
```diff
@@ -267,6 +273,7 @@ Members of the OpenSearch community have built countless plugins for the service
 - [Index State Management]({{site.url}}{{site.baseurl}}/im-plugin/ism/index/)
 - [k-NN]({{site.url}}{{site.baseurl}}/search-plugins/knn/index/)
 - [ML Commons Plugin]({{site.url}}{{site.baseurl}}/ml-commons-plugin/index/)
+- [Neural Search]({{site.url}}{{site.baseurl}}/neural-search/index/)
```
Comment (jmazanec15) [inline _opensearch/install/plugins.md:276]: Neural search should be a search plugin

File: _neural-search-plugin/index.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
+---
+layout: default
+title: Neural Search plugin
+nav_order: 1
+has_children: false
+has_toc: false
+---
+
+The Neural Search plugin is an experimental feature. For updates on the progress of Compare Search Results and other search relevance features, or if you want to leave feedback that could help improve the feature, make an issue in the [Neural Search repo](https://github.com/opensearch-project/neural-search).
```
Comment (jmazanec15) [inline _neural-search-plugin/index.md:9]: Could we also link to this forum post for feedback: https://forum.opensearch.org/t/feedback-neural-search-plugin-experimental-release/11501

File: _neural-search-plugin/index.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+title: Neural Search plugin
+nav_order: 1
+has_children: false
+has_toc: false
+---
+
+The Neural Search plugin is an experimental feature. For updates on the progress of Compare Search Results and other search relevance features, or if you want to leave feedback that could help improve the feature, make an issue in the [Neural Search repo](https://github.com/opensearch-project/neural-search).    
+{: .warning}
+
+The OpenSearch Neural Search plugin enables the integration of Machine Learning (ML) language models into user's search workloads. During ingestion and search, you can provide text through a search query. At query time, the Neural Search plugin automatically transforms the text into vectors to be used in vector search.
```
Comment (jmazanec15) [inline _neural-search-plugin/index.md:12]: > During ingestion and search, you can provide text through a search query

What is meant by this?

File: _opensearch/install/plugins.md (hunk 4 above)
Comment (jmazanec15) [reply _opensearch/install/plugins.md:276]: I see. Just want to confirm that this plugin will be found under Search plugins folder?

File: _neural-search-plugin/index.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
+---
+layout: default
+title: Neural Search plugin
+nav_order: 1
+has_children: false
+has_toc: false
+---
+
+The Neural Search plugin is an experimental feature. For updates on the progress of Compare Search Results and other search relevance features, or if you want to leave feedback that could help improve the feature, join the discussion in the [Neural Search forum](https://forum.opensearch.org/t/feedback-neural-search-plugin-experimental-release/11501).
```
Comment (kolchfa-aws) [inline _neural-search-plugin/index.md:9]:
```suggestion
The Neural Search plugin is an experimental feature. For updates on the progress of the Neural Search plugin, or if you want to leave feedback that could help improve the feature, join the discussion in the [Neural Search forum](https://forum.opensearch.org/t/feedback-neural-search-plugin-experimental-release/11501).    
```

**Review by sean-zheng-amazon — APPROVED** [review]

**Review by jmazanec15 — APPROVED** [review]

**Review by carolxob — APPROVED** [review]

File: _neural-search-plugin/index.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+title: Neural Search plugin
+nav_order: 1
+has_children: false
+has_toc: false
+---
+
+The Neural Search plugin is an experimental feature. For updates on the progress of the Neural Search plugin, or if you want to leave feedback that could help improve the feature, join the discussion in the [Neural Search forum](https://forum.opensearch.org/t/feedback-neural-search-plugin-experimental-release/11501).    
+{: .warning}
+
+The OpenSearch Neural Search plugin enables the integration of Machine Learning (ML) language models into user's search workloads. During ingestion and search, the Neural Search plugin transforms text into vectors. Then, Neural Search uses the transformed vectors in vector-based search."
```
Comment (kolchfa-aws) [inline _neural-search-plugin/index.md:12]: Just noticed after merging that there is an extra quotation mark at the end of this sentence.

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS Please see my comments and changes and let me know if you have any questions. Thanks!

**Pull request merged**